docs/tutorial/mlflow-kubeflow.rst

```diff
@@ -0,0 +1,186 @@
+   juju config dex-auth static-password=user123
+
+Deploy Resource Dispatcher
+--------------------------
+
+Next, let's deploy the resource dispatcher. The resource dispatcher is an optional component which will distribute Kubernetes objects related to MLflow credentials to all user namespaces in Kubeflow. This means that all your Kubeflow users can access the MLflow model registry from their namespaces. To deploy the dispatcher, run the following command:
+
+.. code-block:: bash
+
+   juju deploy resource-dispatcher --channel edge --trust
+
+This will deploy the latest edge version of the dispatcher. See `Resource Dispatcher on GitHub <https://github.com/canonical/resource-dispatcher>`_ for more info. Now we must relate the dispatcher to MLflow:
+
+.. code-block:: bash
+
+   juju relate mlflow-server:secrets resource-dispatcher:secrets
+   juju relate mlflow-server:pod-defaults resource-dispatcher:pod-defaults
+
+Now, at this point, we've deployed MLflow and Kubeflow and we've related them via the resource dispatcher. But that doesn't mean our system is ready yet: Juju will need to download charm data from CharmHub and the charms themselves will take some time to initialise.
+
+So, how do you know when all the charms are ready, then? You can do this using the ``juju status`` command. First, let’s run a basic status command and review the output. Run the following command to print out the status of all the components of Juju:
+
+.. code-block:: bash
+
+   juju status
+
+Review the output for yourself. You should see some summary information, a list of Apps and associated information, and another list of Units and their associated information. Don’t worry too much about what this all means for now. If you’re interested in learning more about this command and its output, see the `Juju Status command <https://juju.is/docs/olm/juju-status>`_.
+
+The main thing we’re interested in at this stage is the statuses of all the applications and units running through Juju. We want all the statuses to eventually become ``active``, indicating that the bundle is ready. Run the following command to keep a watch on the components which are not active yet:
+
+.. code-block:: bash
+
+   watch -c 'juju status --color | grep -E "blocked|error|maintenance|waiting|App|Unit"'
+
+This will periodically run a ``juju status`` command and filter to components which are in a state of ``blocked``, ``error``, ``maintenance`` or ``waiting`` i.e. not ``active``. When this output becomes empty except for the “App” and “Unit” headings, then we know all statuses are active and our system is ready.
+
+Don’t be surprised if some of the components’ statuses change to ``blocked`` or ``error`` every now and then. This is expected behaviour, and these statuses should resolve by themselves as the bundle configures itself. However, if components remain stuck in the same error states for more than 5 minutes or so, consult the troubleshooting steps below.
```

Comment:
If we can't provide something more concrete (and less arbitrary) than "more than 5 minutes or so", let's just drop it. I feel that "if components seem to remain stuck..." conveys the same message in a less hand-wavy way.

Reply:
I agree. Done.